ext/standard/user_filters.c

```diff
@@ -190,9 +190,25 @@ php_stream_filter_status_t userfilter_filter(
 	}
 
 	if (buckets_in->head) {
+		php_stream_bucket *bucket;
+		do {
+			bucket = buckets_in->head;
+			php_stream_bucket_unlink(bucket);
+			php_stream_bucket_delref(bucket);
+		} while (buckets_in->head);
 		php_error_docref(NULL, E_WARNING, "Unprocessed filter buckets remaining on input brigade");
 	}
 
+	/* Filter could've broken contract and added buckets anyway. */
+	if (ret == PSFS_FEED_ME && buckets_out->head) {
```

Comment:
I think I'd need to double check whether this `ret` check is too specific (e.g. do we also need to do anything on FATAL).

Reply:
No, I just checked this and it seems fine with other return values.